datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -229,8 +232,14 @@ protected void translateAndSendBatch(ConsumerRecords<?, ?> records, Instant read
     for (TopicPartition topicPartition : records.partitions()) {
       for (ConsumerRecord<?, ?> record : records.records(topicPartition)) {
         try {
-          if (_autoPausedSourcePartitions.containsKey(topicPartition)) {
-            _logger.warn("Abort sending as {} is auto-paused, rewind offset", topicPartition);
+          boolean skipRecord;
+          synchronized (_sendFailureTopicPartitionExceptionMap) {
+            skipRecord = _autoPausedSourcePartitions.containsKey(topicPartition)
+                || _sendFailureTopicPartitionExceptionMap.containsKey(topicPartition);
+          }
+          if (skipRecord) {
+            _logger.warn("Abort sending as {} {}, rewind offset", topicPartition,
+                _autoPausedSourcePartitions.containsKey(topicPartition) ? "is auto-paused" : "saw a send failure");
```

Comment:
We should set both the values in sychronized above and use them to display.

Reply:
Not sure I understood, which values?

Reply:
I meant: 
```synchronized (_sendFailureTopicPartitionExceptionMap) {
            partitionPaused = _autoPausedSourcePartitions.containsKey(topicPartition)
            sendFailure =   _sendFailureTopicPartitionExceptionMap.containsKey(topicPartition);
          }
          if (partitionPaused || sendFailure) {
            _logger.warn("Abort sending for {} paused: {} sendfailure: {}, rewind offset", topicPartition,
                partitionPaused, sendFailure");```

Reply:
got it, thanks for clarifying!

Reply:
done